cmd/antrea-agent/agent.go

```diff
@@ -187,6 +188,7 @@ func run(o *Options) error {
 			AuthenticationMode: ipsecAuthenticationMode,
 		},
 	}
+	networkConfig.MTUDeduction = networkConfig.CalculateMTUDeduction(enableMulticlusterGW)
```

Comment:
As you made CalculateMTUDeduction() a member of NetworkConfig, how about just passing `enableMulticlusterGW` to `NetworkConfig()` which internally calculates and sets `NetworkConfig.MTUDeduction`, and then we need not `CalculateMTUDeduction()`?

Reply:
Thinking further - how about we add `EnableMulticlusterGW` to be a member of `NetworkConfig`, and even add another member func like `NetworkConfig.NeedsTunnelInterface()`? So we keep the logic in `NetworkConfig`, and need not to pass and check `enableMulticlusterGW` in agent Initializer.

Reply:
Added a new field `EnableMulticlusterGW` to NetworkConfig. Considering it's only one line checking for tunnel interface creation, so I didn't create a new function `NeedsTunnelInterface`, probably we can add it when we have other place to do same check.